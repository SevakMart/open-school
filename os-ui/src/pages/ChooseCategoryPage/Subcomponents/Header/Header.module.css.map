{"version":3,"sources":["Header.module.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","Header.module.css"],"names":[],"mappings":"AAKI;ECJF,yBCDY;EDEZ,kBDIuC;ECHvC,gBDG8C;ECF9C,eDEkD;ECDlD,iBDCuD;EACjD,cGJY;EHKZ,sBAAA;EACA,wBAAA;EACA,YAAA;EACA,qBAAA;AIAR","file":"Header.module.css"}