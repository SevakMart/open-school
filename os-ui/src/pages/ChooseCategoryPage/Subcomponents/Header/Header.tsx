import { useTranslation } from 'react-i18next';
import { Input } from '../../../../component/Input/Input';
import styles from './Header.module.scss';

const Header = ({ changeUrlQueries }:{changeUrlQueries:(title:string)=>void}) => {
  const { t } = useTranslation();
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <h1>{t('string.homePage.header.chooseCategories')}</h1>
      <Input.SearchInput
        changeUrlQueries={(title:string) => changeUrlQueries(title)}
        className={['search', 'search__subcategoriesSearch']}
        placeholder={t('Search by name')}
      />
    </div>
  );
};
export default Header;
