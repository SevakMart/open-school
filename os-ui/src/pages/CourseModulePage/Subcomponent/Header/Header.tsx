import NavbarOnSignIn from '../../../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import styles from './Header.module.scss';

const Header = () => {
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <NavbarOnSignIn />
    </div>
  );
};
export default Header;
