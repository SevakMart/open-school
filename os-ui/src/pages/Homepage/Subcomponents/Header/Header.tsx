import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { Types } from '../../../../types/types';
import { openModal } from '../../../../redux/Slices/PortalOpenStatus';
import Navbar from '../../../../component/Navbar-Component/Navbar/Navbar';
import HomepageMainImage from '../../../../assets/svg/HomepageMainImage.svg';
import Button from '../../../../component/Button/Button';
import styles from './Header.module.scss';

const HomepageHeader = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const {
    mainContent, leftContent, rightContent, buttonContainer, leftContentSignInBtn,
  } = styles;

  const handleSignUp = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_UP }));
  };

  const handleSignIn = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_IN }));
  };

  return (
    <>
      <Navbar />
      <div className={mainContent}>
        <div className={leftContent}>
          <h1>{t('string.homePage.header.educationalPlatform')}</h1>
          <p>{t('Join our community of mentors and help shape')}</p>
          <p>{t('the future of tech talent')}</p>
          <div className={buttonContainer}>
            <>
              <Button.SignUpButton className={['headerButton', 'headerButton__signUp']} onClick={handleSignUp}>
                {t('button.homePage.signUp')}
              </Button.SignUpButton>
              <button type="button" onClick={handleSignIn} className={leftContentSignInBtn}>
                {t('button.homePage.signIn')}
              </button>
            </>
          </div>
        </div>
        <div className={rightContent}>
          <img src={HomepageMainImage} alt="Education platform logo" />
        </div>
      </div>
    </>
  );
};
export default HomepageHeader;
