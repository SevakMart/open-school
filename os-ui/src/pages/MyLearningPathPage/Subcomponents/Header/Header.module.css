.mainHeader {
  display: inline-block;
  margin: 64px 0 56px 88px;
  box-sizing: border-box;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 48px;
  line-height: 59px;
  color: #222427;
}

.courseNavigationBar {
  margin-left: 88px;
  box-sizing: border-box;
}
.courseNavigationBar .activeNav {
  display: inline-block;
  margin: 0 24px 0 0;
  cursor: pointer;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #222427;
  border-bottom: 2px solid #222427;
}
.courseNavigationBar .nonActiveNav {
  margin: 0 24px 0 0;
  display: inline-block;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B;
  cursor: pointer;
}
.courseNavigationBar .nonActiveNav:hover {
  color: #222427;
}/*# sourceMappingURL=Header.module.css.map */