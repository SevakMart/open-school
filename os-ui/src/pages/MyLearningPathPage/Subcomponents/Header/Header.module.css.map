{"version":3,"sources":["Header.module.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","Header.module.css"],"names":[],"mappings":"AAKA;EACI,qBAAA;EACA,wBAAA;EACA,sBAAA;ECPF,yBCDY;EDEZ,kBDOmC;ECNnC,gBDM0C;ECL1C,eDK8C;ECJ9C,iBDImD;EACjD,cGPgB;ACOpB;;AJEA;EACI,iBAAA;EACA,sBAAA;AICJ;AJAI;EACI,qBAAA;EACA,kBAAA;EACA,eAAA;ECjBN,yBCDY;EDEZ,kBDiBuC;EChBvC,gBDgB8C;ECf9C,eDekD;ECdlD,iBDcuD;EACjD,cGjBY;EHkBZ,gCAAA;AIMR;AJJI;EACI,kBAAA;EACA,qBAAA;ECxBN,yBCDY;EDEZ,kBDwBuC;ECvBvC,gBDuB8C;ECtB9C,eDsBkD;ECrBlD,iBDqBuD;EACjD,cGzBU;EH0BV,eAAA;AIUR;AJTQ;EACI,cG3BQ;ACsCpB","file":"Header.module.css"}