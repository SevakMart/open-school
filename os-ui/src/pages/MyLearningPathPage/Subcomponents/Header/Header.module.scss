@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";
@import "../../../../styles/Mixins/FontMixin.scss";


.mainHeader {
    display: inline-block;
    margin: 64px 0 56px 88px;
    box-sizing: border-box;
    @include fontStyle($primary-font,normal,800,48px,59px);
    color: $header-main-color;
}
.courseNavigationBar{
    margin-left:88px;
    box-sizing: border-box;
    .activeNav {
        display:inline-block;
        margin:0 24px 0 0;
        cursor:pointer;
        @include fontStyle($primary-font,normal,600,20px,24px);
        color:$header-main-color;
        border-bottom: 2px solid $header-main-color ;
    }
    .nonActiveNav {
        margin:0 24px 0 0;
        display:inline-block;
        @include fontStyle($primary-font,normal,500,16px,24px);
        color:$main-text-color;
        cursor:pointer;
        &:hover {
            color:$header-main-color;
        }
    }
}