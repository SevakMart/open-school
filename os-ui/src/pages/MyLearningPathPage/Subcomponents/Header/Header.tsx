import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import NavbarOnSignIn from '../../../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import { getUserEnrolledCourseByCourseStatus } from '../../../../redux/Slices/UserEnrolledCourseByCourseStatus';
import styles from './Header.module.scss';

enum LearningPathNav {
    All='All',
    InProgress='InProgress',
    Completed='Completed'
}
/* eslint-disable max-len */
const Header = ({ userId, token }:{userId:number, token:string}) => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const [activeNavType, setActiveNavType] = useState(LearningPathNav.All);
  const {
    mainHeader, courseNavigationBar, activeNav, nonActiveNav,
  } = styles;

  const handleNavigation = (e:React.SyntheticEvent) => {
    setActiveNavType((e.target as HTMLParagraphElement).dataset.testid as any);
  };

  useEffect(() => {
    switch (activeNavType) {
      case LearningPathNav.All:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token }));
        break;
      case LearningPathNav.InProgress:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token, params: { courseStatusId: 1 } }));
        break;
      case LearningPathNav.Completed:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token, params: { courseStatusId: 2 } }));
        break;
    }
  }, [activeNavType]);

  return (
    <>
      <NavbarOnSignIn />
      <h1 className={mainHeader}>{t('string.myLearningPaths.mainHeaderTitle')}</h1>
      <nav className={courseNavigationBar}>
        <p className={activeNavType === 'All' ? activeNav : nonActiveNav} data-testid={LearningPathNav.All} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.all')}</p>
        <p className={activeNavType === 'InProgress' ? activeNav : nonActiveNav} data-testid={LearningPathNav.InProgress} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.inProgress')}</p>
        <p className={activeNavType === 'Completed' ? activeNav : nonActiveNav} data-testid={LearningPathNav.Completed} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.completed')}</p>
      </nav>
    </>
  );
};
export default Header;
