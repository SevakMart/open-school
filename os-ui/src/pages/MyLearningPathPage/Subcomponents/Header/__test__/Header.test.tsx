import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import Header from '../Header';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => ({
    search: '',
  }),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for Header', () => {
  test('Create snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Header
          userId={1}
          token="123"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
