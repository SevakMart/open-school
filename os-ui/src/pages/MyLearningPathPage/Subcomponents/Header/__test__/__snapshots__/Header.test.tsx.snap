// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for Header Create snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="navMainContent"
    >
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.allLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.myLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.mentors
      </p>
      <div
        class="bookmarkIconStyle"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="font-size: 20px; cursor: pointer; fill: none;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
          />
        </svg>
      </div>
      <img
        alt="Notification"
        class="notificationIcon"
        src="Notification.svg"
      />
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          class="avatarLogo"
          src="https://reactjs.org/logo-og.png"
        />
        <img
          alt="Arrow down Icon"
          class="arrowDownIcon"
          src="Arrow-down.svg"
        />
      </div>
      <div />
    </div>
  </nav>
  <h1
    class="mainHeader"
  >
    string.myLearningPaths.mainHeaderTitle
  </h1>
  <nav
    class="courseNavigationBar"
  >
    <p
      class="activeNav"
      data-testid="All"
    >
      string.myLearningPaths.courseStatus.all
    </p>
    <p
      class="nonActiveNav"
      data-testid="InProgress"
    >
      string.myLearningPaths.courseStatus.inProgress
    </p>
    <p
      class="nonActiveNav"
      data-testid="Completed"
    >
      string.myLearningPaths.courseStatus.completed
    </p>
  </nav>
</DocumentFragment>
`;
